cupy/core/_routines_math.pyx

```diff
@@ -192,15 +342,20 @@ def _add_scan_blocked_sum_kernel(dtype):
         unsigned int idxAdd = idxBase - 1;
 
         if(idxAdded < n){
-            src_dst[idxAdded] += src_dst[idxAdd];
+            src_dst[idxAdded] ${op}= src_dst[idxAdd];
         }
     }
-    """).substitute(name=name, dtype=dtype)
+    """).substitute(name=name, dtype=dtype, op=ops[op])
     module = compile_with_cache(source)
     return module.get_function(name)
 
 
-cdef ndarray scan(ndarray a, ndarray out=None):
+cdef ndarray scan(
+        ndarray a, op=scan_op.SCAN_SUM, dtype=None, ndarray out=None):
+    return _scan_op(a, op, dtype, out)
+
+
+cdef ndarray _scan_op(ndarray a, op, dtype, ndarray out=None):
```

Comment:
I feel confused that this function name similar to `scan_op` enum type name.

Reply:
How about writing the implementation in `scan`?

Reply:
Is it ok to maintain the name in "scan"?